commerce-machine-learning-impl/src/main/java/com/liferay/commerce/machine/learning/internal/forecast/CommerceAccountCommerceMLForecastServiceImpl.java

```diff
@@ -49,6 +49,30 @@
 	extends BaseCommerceMLForecastServiceImpl<CommerceAccountCommerceMLForecast>
 	implements CommerceAccountCommerceMLForecastService {
 
+	@Override
+	public CommerceAccountCommerceMLForecast
+			addCommerceAccountCommerceMLForecast(
+				CommerceAccountCommerceMLForecast
+					commerceAccountCommerceMLForecast)
+		throws PortalException {
+
+		long commerceMLForecastId = getHash(
+			commerceAccountCommerceMLForecast.getCommerceAccountId(),
+			commerceAccountCommerceMLForecast.getPeriod(),
+			commerceAccountCommerceMLForecast.getScope(),
+			commerceAccountCommerceMLForecast.getTarget(),
+			commerceAccountCommerceMLForecast.getTimestamp());
```

Comment:
Why this solution for the ID? Do we specifically want the hash deterministic -- i.e. the same field values should always yield the same ID?

Reply:
Yes, bear in mind those id are computed outside the portal where there is no counter facility. We also need to be able to retrieve/ re-generate the ID when updating the record so the id must be a function of a well known set of fields and cannot be a generally increasing monotonic number

Reply:
Got it, alright. Nice solution then :)